### PR TITLE
Peridot soul: fix accidental event global

### PR DESCRIPTION
**Pull request by modernserf**
This block of code was using [window.event](https://developer.mozilla.org/en-US/docs/Web/API/Window/event), which is _usually_ what you want except when it isn't.

I found this while chasing down [this bug](https://sentry.io/organizations/fog-creek-software/issues/928442716) but it appears to be unrelated.

### CODE REVIEW

**Comment by modernserf** [comment]
its ... not impossible that this could be related to the too much recursion on file upload bug?

**Comment by GregWeil** [comment]
We only use progress promises in the s3 uploader. If you're feeling bold, removing them entirely would probably make this easier to reason about.

In the mean time this change looks sound on its own, the one place we use the notify function does give it an event object.

**Comment by clottman** [comment]
I think this is good to merge/deploy. And perhaps removing the progress promise can be a future ticket/part of the infinite recursion investigation.

**Comment by modernserf** [comment]
Updated this PR to:
- re-enable linting on this file (!)
- enable a lint rule that will catch usage of the `event` global

File: src/utils/progress-promise.js [hunk 1]
```diff
@@ -21,10 +19,10 @@ if (Promise.prototype.finally == null) {
 }
 
 if (Promise.prototype._notify == null) {
-  Promise.prototype._notify = function(event) {
+  Promise.prototype._notify = function _notify(event) {
     return this._progressHandlers.forEach((handler) => {
       try {
-        return handler(event);
```
Comment (modernserf) [inline src/utils/progress-promise.js]: for consistent return values; forEach doesn't use return values anyways

File: src/utils/progress-promise.js [hunk 2]
```diff
@@ -1,11 +1,9 @@
-/* eslint-disable */
-
-export default ProgressPromise;
+/* eslint-disable no-extend-native, no-underscore-dangle */
 
 // Extend promises with `finally`
 // From: https://github.com/domenic/promises-unwrapping/issues/18
 if (Promise.prototype.finally == null) {
-  Promise.prototype.finally = function(callback) {
+  Promise.prototype.finally = function _finally(callback) {
```
Comment (modernserf) [inline src/utils/progress-promise.js]: for func-names

File: src/utils/progress-promise.js [hunk 3]
```diff
@@ -44,21 +42,23 @@ if (Promise.prototype.progress == null) {
...
-        ? p._progressHandlers.forEach((handler) => {
-            try {
-              return handler(event);
-            } catch (error) {
-              // empty
-            }
-          })
-        : undefined;
+  const p = new Promise((resolve, reject) => {
+    const notify = (event) => {
```
Comment (modernserf) [inline src/utils/progress-promise.js]: add proper `event` variable

File: src/utils/progress-promise.js [hunk 4]
```diff
@@ -44,21 +42,23 @@ if (Promise.prototype.progress == null) {
...
+      if (p._progressHandlers == null) {
```
Comment (modernserf) [inline src/utils/progress-promise.js]: fix indentation (pathological case with prettier vs eslint rules)

File: src/utils/progress-promise.js [hunk 5]
```diff
@@ -21,10 +19,10 @@ if (Promise.prototype.finally == null) {
 }
 
 if (Promise.prototype._notify == null) {
-  Promise.prototype._notify = function(event) {
+  Promise.prototype._notify = function _notify(event) {
```
Comment (sarahzinger) [inline src/utils/progress-promise.js]: for my own brain, does this change do something or is it a stylistic thing?

Comment (modernserf) [reply src/utils/progress-promise.js]: it potentially changes what stack traces look like, but its mostly just to please eslint

**Comment by sarahzinger** [comment]
hmm seems legit to me! not sure why the runner has failed

**Pull request merged**